Sources/MongoSwift/MongoClient.swift

```diff
@@ -72,8 +72,19 @@ public struct ClientOptions: CodingStrategyProvider, Decodable {
     /// Specifies a WriteConcern to use for the client.
     public var writeConcern: WriteConcern?
 
-    private enum CodingKeys: CodingKey {
-        case retryWrites, retryReads, readConcern, writeConcern
+    private enum CodingKeys: String, CodingKey {
+        case retryWrites, retryReads, readConcern, writeConcern, w, readConcernLevel, mode = "readPreference"
+    }
+
+    public init(from decoder: Decoder) throws {
```

Comment:
This `Decodable` initializer was needed because `clientOptions` in the transactions spec tests have a weird format that can't be decoded with the default initializer. Here's an example from **transaction-options.json**:
```
{
      "description": "transaction options inherited from client",
      "clientOptions": {
        "w": 1,
        "readConcernLevel": "local"
      },
      "operations":
      ...
}
```
I agree that it's probably not good practice to add this decoding logic into the public API. I've therefore created a new struct called `TestClientOptions` in **SpecTest.swift** that decodes `clientOptions` in the spec tests properly and then has a function that converts it into a corresponding `ClientOptions` struct.

Reply:
bah, that's unfortunate. Could we not add an extension to `ClientOptions` into `CodableExtensions.swift` instead of introducing the new type?

Reply:
I tried doing this but since `ClientOptions` already conforms to `Decodable`, you cannot provide another initializer of the form `init(from decoder: Decoder)`.

Reply:
Oh I see. I think we can just remove the `Codable` conformance in `ClientOptions` declaration then--any "options" codable conformance exists right now only for testing purposes. We figured it saved us from enough boilerplate and that users probably wouldn't try to actually use the conformance, so it was worth it to bleed into the API. I think it must have been an earlier version of Swift or something that required us to conform in the driver rather than the tests, or maybe we just never tried lol.

Reply:
So what I meant by this was to remove the `Decodable` conformance in [this line](https://github.com/mongodb/mongo-swift-driver/blob/master/Sources/MongoSwift/MongoClient.swift#L6). Right now, that declaration is causing the compiler to "synthesize" a `Decodable` conformance in the driver code, which is why you're seeing the error complaining about multiple decodable conformances. Once that's removed, you can add it as an extension in the tests.

Reply:
Ah I think I missed addressing this comment. I've removed `TestClientOptions` and simply written a decodable extension of `ClientOptions`.